### PR TITLE
Fix debian gnupg installation

### PR DESCRIPTION
**Pull request by mcalmer**
## What does this PR change?

Install gnupg on debian after trusting the gpg key which signed the Uyuni metadata.
Moving also res key trust to gpg-key.sls file.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- No tests: already covered

- [x] **DONE**

## Links

Fixes https://github.com/uyuni-project/uyuni/issues/3624
Tracks https://github.com/SUSE/spacewalk/pull/14965

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by juliogonzalez** [comment]
@mcalmer add the `merge-candidate` if we want this merged for GMC and Uyuni-2021.05.

I'd merge it now, but you don't have reviewers, so not sure that means this is not ready yet.

**Comment by mcalmer** [comment]
@juliogonzalez no idea if this fix the problem. I wait for a community to answer. Do not merge yet.

File: susemanager/susemanager.changes
Comment (juliogonzalez) [inline susemanager/susemanager.changes:1]: Are the python3 dependencies for gnupug? Or are they dependencies for  `python3-cryptodome`?

**Review by juliogonzalez — APPROVED** [review]
Just one question :-)

File: susemanager/susemanager.changes
Comment (mcalmer) [reply susemanager/susemanager.changes:1]: The only new python package is python3-cryptodome. The list looked like sorted alphabetically so I added gnupg and its dependecies and run a "sort" on the list. This moved also other packages around.

Comment (juliogonzalez) [reply susemanager/susemanager.changes:1]: True! Sorry for the noise :-)

**Review by agraul — APPROVED** [review]
I think everything is correct, just one question. `python3-pycryptodome` is only in 18.04 Universe. What's the current Universe status? Is it mandatory or do we need to ship `python3-pycryptodome` for 18.04 ourselves?

**Comment by juliogonzalez** [comment]
> I think everything is correct, just one question. `python3-pycryptodome` is only in 18.04 Universe. What's the current Universe status? Is it mandatory or do we need to ship `python3-pycryptodome` for 18.04 ourselves?

@mcalmer WARNING! Then this means we need to do more after merging this PR. Let's talk at Rocket.Chat

**Comment by mcalmer** [comment]
> I think everything is correct, just one question. `python3-pycryptodome` is only in 18.04 Universe. What's the current Universe status? Is it mandatory or do we need to ship `python3-pycryptodome` for 18.04 ourselves?

salt 3002 package has it as dependency. Is this a bug or do I mix it with recommends. @agraul can you check the deps of the Ubuntu package (salt-common)

**Comment by mcalmer** [comment]
```
Package: salt-common
Source: salt
Version: 3002.2+ds-1+27.37.1
Architecture: all
Maintainer: Debian Salt Team <pkg-salt-team@lists.alioth.debian.org>
Installed-Size: 30694
Depends: python3-apt, python3-dateutil, python3-jinja2, python3-msgpack (>= 0.4), python3-pkg-resources, python3-requests, python3-yaml, python3-systemd, python3-psutil, python3-distro, python3-markupsafe, python3-pycryptodome, python3-zmq, python3:any (>= 3.3.2-2~)
```
This is from Ubuntu 18.04 client tools beta repo

**Comment by agraul** [comment]
> ```
> Package: salt-common
> Source: salt
> Version: 3002.2+ds-1+27.37.1
> Architecture: all
> Maintainer: Debian Salt Team <pkg-salt-team@lists.alioth.debian.org>
> Installed-Size: 30694
> Depends: python3-apt, python3-dateutil, python3-jinja2, python3-msgpack (>= 0.4), python3-pkg-resources, python3-requests, python3-yaml, python3-systemd, python3-psutil, python3-distro, python3-markupsafe, python3-pycryptodome, python3-zmq, python3:any (>= 3.3.2-2~)
> ```
> 
> This is from Ubuntu 18.04 client tools beta repo

I'm not sure which Python "crypto" libraries usable by Salt, in our specfiles we use `python3-M2Crypto` if we can and fall back to `python3-pycrypto`. But one of them is required. Since QA tested with `python3-pycryptodome`, I think we should start shipping it in client tools.

**Pull request merged**
